src/main/java/no/nav/brregstub/mapper/RolleoversiktMapper.java

```diff
@@ -95,7 +147,25 @@ private static Melding mapTilMelding(RolleoversiktTo to) {
             }
         }
         return roller;
+    }
 
+    private static Melding.Roller mapTilRoller(RsRolleoversikt rsRolleoversikt) {
+        var roller = new Melding.Roller();
+        if (rsRolleoversikt.getEnheter() != null) {
+            int count = 1;
+            for (RsRolle rsRolle : rsRolleoversikt.getEnheter()) {
+                var enhet = new Enhet();
+                enhet.setRolleBeskrivelse(mapTilRollebeskrivelse(rsRolle.getRolle()));
+                enhet.setNr(count);
+                count++;
+                enhet.setNavn(mapTilNavntype(rsRolle.getForetaksNavn()));
+                enhet.setOrgnr(mapTilOrganisasjonsNummer(rsRolle.getOrgNr()));
+                enhet.setAdresse(mapTilAdresseEnhet(rsRolle));
+                enhet.setRegistreringsDato(localDateToXmlGregorianCalendar(rsRolle.getRegistreringsdato()));
+                roller.getEnhet().add(enhet);
+            }
```

Comment:
Hvis du lager count som AtomicInteger, har forEach løkke og en builder for Enhet kan dette gjøres som to statements.
(Tror jeg da)

Reply:
En del av klassene her er automatisk opprettet fra xsder fra brønnøysund, så ville ikke fikle for mye med dem